code/modules/clothing/spacesuits/flightsuit.dm

```diff
@@ -492,6 +520,38 @@
 		losecontrol(move = FALSE)
 	crashing = FALSE
 
+/obj/item/device/flightpack/proc/door_hit(obj/structure/mineral_door/door)
+	spawn()
+		door.Open()
+	wearer.forceMove(get_turf(door))
+	wearer.visible_message("<span class='boldnotice'>[wearer] rolls to their sides and slips past [door]!</span>")
+
+
+/obj/item/device/flightpack/proc/crash_grille(obj/structure/grille/target)
+	target.hitby(wearer)
+	target.take_damage(60, BRUTE, "melee", 1)
+	if(wearer.Move(target.loc))
+		wearer.visible_message("<span class='warning'>[wearer] smashes straight past [target]!</span>")
+
+/obj/item/device/flightpack/proc/airlock_hit(obj/machinery/door/A)
+	var/pass = 0
+	if(A.density)	//Is it closed?
+		pass += A.locked
+		pass += A.stat	//No power, no automatic open
+		pass += A.emagged
+		if(A.requiresID())
+			if((!A.allowed(wearer)) && !A.emergency)
+				pass += 1
+	else
+		return pass
+	if(!pass)
+		spawn()
```

Comment:
Bugs out if I don't I assume it waits for it to open completely before letting you through which ends up in you teleporting back to the door after you fly through.